parse.py

```diff
@@ -271,6 +271,46 @@ def date_convert(
     return d
 
 
+dt_format_to_regex = {symbol: "[0-9]{2}" for symbol in "ymdIMSUW"}
+dt_format_to_regex.update({"-" + symbol: "[0-9]{1,2}" for symbol in "ymdIMS"})
```

Comment:
For some reason I get a problem with these:
```
>>> datetime(2023, 1, 1).strftime("%Y/%-m/%-d")
'2023/1/1'
```
But
```
>>> parse("{:%Y/%-m/%-d}", "2023/1/1")
ValueError: '-' is a bad directive in format '%Y/%-m/%-d'
```

Reply:
Hmmm I'll take a look tomorrow

Reply:
This is now fixed. It turns out strptime does not use the negative sign (I'm not quite sure why I thought it did). In strftime, %d outputs a zero-padded number e.g. "01". For strptime, %d matches a zero-padded number and can also match a non-zero padded number e.g. "1".

Reply:
Now the following works:

```python
>>> parse("{:%Y/%m/%d}", "2023/01/01")
<Result (datetime.datetime(2023, 1, 1, 0, 0),) {}>
```

Reply:
That's better.  %j should get the same treatment, because it has the same issue (i.e. "%-j" doesn't work with strptime, and %j doesn't care if the number is zero-padded or not).

After removing the mapping for `"-j"`, you can also remove the `re.escape` since the remaining characters are all letters and don't need escaping.